sdclientapi/__init__.py

```diff
@@ -118,19 +144,21 @@ def authenticate(self, totp="") -> bool:
             )
         except json.decoder.JSONDecodeError:
             raise BaseError("Error in parsing JSON")
-        if not "expiration" in token_data:
+        if "expiration" not in token_data:
             raise AuthError("Authentication error")
-        self.token = token_data
+        self.token = token_data["token"]
+        self.token_expiration = datetime.strptime(
+            token_data["expiration"], "%Y-%m-%dT%H:%M:%S.%fZ"
+        )
         self.update_auth_header()
 
-        return True
-
-    def update_auth_header(self):
-        self.auth_header = {
-            "Authorization": "Token " + self.token["token"],
-            "Content-Type": "application/json",
-            "Accept": "application/json",
-        }
+    def update_auth_header(self) -> None:
+        if self.token is not None:
```

Comment:
`mypy` yells at this because these two statements are not equivalent. Also, there is never a case where the token will be an empty string, so this works too.

Reply:
https://github.com/freedomofpress/securedrop-sdk/pull/63/files#diff-922f10157cf861460ae77dc5c7dc980eR76 this is where it should be an empty string to represent `False` or in this case, the value to represent that the token has not been setup yet.

Reply:
Same should be with `headers`, `something is not None` should be avoided in Python code.